### ISSUE TITLE
Figure out how to ensure GPG Key installs work properly on a restrictive network

### ISSUE THREAD

**Issue by jeffgeorge** [clwdev/precip#39]
GPG Keys are installed by communicating with -- for instance -- `keyserver.ubuntu.com` over port 80, but they're then _sent_ over port 11731. If port 11731 is blocked Very Bad Things occur, like PHP 5.5 being installed from Ubuntu rather than ondrej's PPA, and Percona will _super_ refuse to install properly.

We have a few ways around this, like there's a way to [bounce through a proxy](http://askubuntu.com/questions/53146/how-do-i-get-add-apt-repository-to-work-through-a-proxy), which we could provide . . or we could just ensure that a packed version of Precip comes with the keys all baked-in.

Either way, when this fails the messaging really sucks, so we should find a way to fix it.

**Comment by jeffgeorge** [comment]
Pull Request #40 introduces "Packaging Mode", which starts us on the path to fix this issue. Once we introduce a conditional switch to use the partially-provisioned packaged box instead of the puppetlabs box this issue should be resolved.

**Comment by vitalie-cracan** [comment]
I get a slightly different issue related to GPG key which seem to prevent the provisioning form running successfully, it fails on all apt packages with an error like this:

```sh
DEBUG ssh: stderr: [1;31mError: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install git' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  git-man liberror-perl
Suggested packages:
  git-daemon-run git-daemon-sysvinit git-doc git-el git-email git-gui gitk
  gitweb git-arch git-bzr git-cvs git-mediawiki git-svn
The following NEW packages will be installed:
  git git-man liberror-perl
0 upgraded, 3 newly installed, 0 to remove and 127 not upgraded.
Need to get 3400 kB/3421 kB of archives.
After this operation, 21.9 MB of additional disk space will be used.
Err http://us.archive.ubuntu.com/ubuntu/ trusty-updates/main git-man all 1:1.9.1-1ubuntu0.2
  404  Not Found [IP: 91.189.91.23 80]
Err http://security.ubuntu.com/ubuntu/ trusty-security/main git-man all 1:1.9.1-1ubuntu0.2
  404  Not Found [IP: 91.189.91.23 80]
Err http://security.ubuntu.com/ubuntu/ trusty-security/main git amd64 1:1.9.1-1ubuntu0.2
  404  Not Found [IP: 91.189.91.23 80]
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git-man_1.9.1-1ubuntu0.2_all.deb  404  Not Found [IP: 91.189.91.23 80]

E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git_1.9.1-1ubuntu0.2_amd64.deb  404  Not Found [IP: 91.189.91.23 80]

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?[0m
[1;31mError: /Stage[main]/Precip/Package[git]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install git' returned 100: Reading package lists...
```

Then when manually running:
`sudo apt-get update`

The following error pops-up:

```sh
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: http://apt.puppetlabs.com trusty Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 7F438280EF8D349F
```

To fix, I manually ran this on Precip box:
```sh
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 7F438280EF8D349F
sudo apt-get update
```

Then, `vagrant provision` on host machine.

**Comment by jeffgeorge** [comment]
Unfortunately we can't really test the original issue anymore because we no longer have access to a restricted network. @en-vcracan, the similar issue you encountered here _should_ be resolved on the `ubuntu-xenial` branch, see PR #53.
